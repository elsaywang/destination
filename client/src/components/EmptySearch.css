:local(.noResultMessage) {
    margin: -0.75em 0 0 1.85em;
}
:local(.exploreMessage) {
    margin: -0.75em 0 0 1.5em;
}
:local(.noResultImage) {
    width: 90px;
    height: 80px;
    margin-left: -8px;
}
:local(.exploreImage) {
    width: 190px;
    height: 101px;
}
