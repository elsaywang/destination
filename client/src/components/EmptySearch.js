import React from 'react';
import PropTypes from 'prop-types';
import EmptyPlaceholder from './common/EmptyPlaceholder';
import styles from './EmptySearch.css';
import Explore from '../images/explore.svg';
import NoResult from '../images/noResult.svg';
import { getEmptyOptions } from '../constants/emptyOptions';
import { getStaticUrl } from '../lib/getStaticUrl';

function EmptySearch({ variant, className }) {
    const getMessageStyle = () => {
        switch (variant) {
            case 'explore':
                return styles.exploreMessage;
            case 'noResult':
                return styles.noResultMessage;
            case 'errorFetching':
                return styles.noResultMessage;
            default:
                return '';
        }
    };
    const getImageClass = () => {
        switch (variant) {
            case 'explore':
                return styles.exploreImage;
            case 'noResult':
                return styles.noResultImage;
            case 'errorFetching':
                return styles.noResultImage;
            default:
                return '';
        }
    };
    const getImageSrc = () => {
        switch (variant) {
            case 'explore':
                return `${getStaticUrl()}${Explore}`;
            case 'noResult':
                return `${getStaticUrl()}${NoResult}`;
            case 'errorFetching':
                return `${getStaticUrl()}${NoResult}`;
            default:
                return '';
        }
    };

    return (
        <EmptyPlaceholder
            className={className}
            title={getEmptyOptions(variant).title}
            message={getEmptyOptions(variant).message}
            messageStyle={getMessageStyle()}>
            <img
                src={getImageSrc()}
                data-test={getEmptyOptions(variant).dataTest}
                className={getImageClass()}
                alt={getEmptyOptions(variant).imageAlt}
            />
        </EmptyPlaceholder>
    );
}

EmptySearch.propTypes = {
    variant: PropTypes.oneOf(['explore', 'noResult']).isRequired,
};

export default EmptySearch;
