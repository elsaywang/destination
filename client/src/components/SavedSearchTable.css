:local(.noResults) {
    padding: 64px 0;
    background: #fff;
    text-align: center;
}
