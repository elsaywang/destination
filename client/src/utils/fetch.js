import { getCsrfToken } from '../lib/getCsrfToken';

export const getOptionsWithAAMAuth = (options = { headers: {} }) => ({
    ...options,
    headers: {
        'content-type': 'application/json',
        ...options.headers,
        'AAM-CSRF-Token': getCsrfToken(),
    },
    credentials: 'same-origin',
});

export default function(url, options) {
    return fetch(url, getOptionsWithAAMAuth(options));
}
