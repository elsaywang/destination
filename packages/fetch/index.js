export const getOptionsWithAAMAuth = (options = {
  headers: {}
}) => ({ ...options,
  headers: {
    'content-type': 'application/json',
    ...options.headers,
    'AAM-CSRF-Token': document.querySelector('meta[name="_tk"]').content,
    'AAM-Disable-Escaping-HTML': true
  },
  credentials: 'same-origin'
});

const fetch = (url, options) => window.fetch(url, getOptionsWithAAMAuth(options));

export default fetch;
