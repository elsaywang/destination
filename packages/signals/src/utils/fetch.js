import { getCsrfToken } from '../lib/getCsrfToken';

export const getOptionsWithAAMAuth = (options = { headers: {} }) => ({
    ...options,
    headers: {
        'content-type': 'application/json',
        ...options.headers,
        'AAM-CSRF-Token': getCsrfToken(),
        'AAM-Disable-Escaping-HTML': true,
    },
    credentials: 'same-origin',
});

const fetch = (url, options) => window.fetch(url, getOptionsWithAAMAuth(options));

export default fetch;
