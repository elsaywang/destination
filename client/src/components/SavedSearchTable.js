import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Wait from '@react/react-spectrum/Wait';
import SignalsTable from './SignalsTable';
import EmptySearch from './EmptySearch';
import { fetchSignals } from '../utils/fetchSignals';
import styles from './SavedSearchTable.css';

class SavedSearchTable extends Component {
    state = {
        tableResults: {},
        error: false,
        hasSearched: false,
    };

    async componentDidMount() {
        const { savedSearch } = this.props;
        const response = await fetchSignals({ search: savedSearch });

        if (response.ok) {
            const results = await response.json();

            this.setState({
                tableResults: results,
                error: false,
                hasSearched: true,
            });
        } else {
            this.setState({
                tableResults: {},
                error: true,
                hasSearched: true,
            });
        }
    }

    renderTable() {
        const { tableResults } = this.state;
        const {
            isAdvancedSearchEnabled,
            allowsSelection,
            canCreateTraits,
            savedSearch,
        } = this.props;
        const hasSearchResults = !this.state.error && tableResults.list && tableResults.list.length;
        const totalKeyValuePairs = savedSearch.keyValuePairs.length;
        const withResults = hasSearchResults && !this.state.error;

        if (withResults) {
            return (
                <SignalsTable
                    isLoaded={hasSearchResults}
                    results={tableResults.list}
                    totalKeyValuePairs={totalKeyValuePairs}
                    canCreateTraits={canCreateTraits}
                    isAdvancedSearchEnabled={isAdvancedSearchEnabled}
                    allowsSelection={allowsSelection}
                />
            );
        }

        return (
            <EmptySearch
                className={styles.noResults}
                variant={this.state.error ? 'errorFetching' : 'noResult'}
            />
        );
    }

    render() {
        return this.state.hasSearched ? this.renderTable() : <Wait />;
    }
}

SavedSearchTable.propTypes = {
    savedSearch: PropTypes.object,
    getResultsBySavedSearch: PropTypes.func,
    isAdvancedSearchEnabled: PropTypes.bool,
    allowsSelection: PropTypes.bool,
    canCreateTraits: PropTypes.bool,
};

export default SavedSearchTable;
