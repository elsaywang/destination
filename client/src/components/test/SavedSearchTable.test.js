import React from 'react';
import { shallow } from 'enzyme';
import SavedSearchTable from '../SavedSearchTable';
import EmptySearch from '../EmptySearch';
import fetch from '../../utils/fetch';

jest.mock('../../utils/fetch');
fetch.mockImplementation(() => ({ json: () => new Promise(res => ({})) }));

describe('<SavedSearchTable/> component', () => {
    const mockFn = jest.fn();
    const props = {
        savedSearch: {
            name: "Margot O'Hara",
            keyValuePairs: [
                {
                    key: 'k-user-centric',
                    operator: '>=',
                    value: 31399,
                    id: 0,
                },
            ],
            source: {
                dataSourceIds: 72599,
                reportSuiteIds: null,
                sourceType: 'REALTIME',
            },
            minEventFires: 46276,
            signalStatus: 'USED',
            startDate: '2018-04-22T20:41:09.693Z',
            endDate: '2018-04-23T10:37:26.452Z',
        },
        isAdvancedSearchEnabled: false,
        allowsSelection: false,
    };

    const wrapper = shallow(<SavedSearchTable {...props} getResultsBySavedSearch={mockFn} />);

    describe('rendering', () => {
        const newState = {
            tableResults: {},
            error: false,
            hasSearched: false,
        };

        wrapper.setState({ ...newState });

        const initialWrapper = wrapper.dive();

        it('matches snapshot', () => {
            expect(initialWrapper).toMatchSnapshot();
        });

        it('does not render <SignalsTable>', () => {
            const signalsTable = initialWrapper.findWhere(
                node => node.props().dataTest === 'signals-table',
            );

            expect(signalsTable.exists()).toBeFalsy();
        });
    });

    describe('state changes based on lifecyle events without error', () => {
        const newState = {
            tableResults: {
                pid: 1194,
                startDate: 1530662400000,
                endDate: 1531332629645,
                list: [
                    {
                        keyValuePairs: [
                            {
                                key: 'evar18',
                                value: 'CMS18',
                            },
                        ],
                        source: {
                            dataSourceIds: null,
                            reportSuiteIds: null,
                            sourceType: null,
                        },
                        totalCount: 93859,
                        percentageChange: 0.27783491545226535,
                        includedInTraits: null,
                        signalStatus: 'UNUSED',
                    },
                ],
                pageSize: 20,
                page: 0,
                total: 20,
                analyticsServiceAvailable: true,
            },
            error: false,
            hasSearched: true,
        };

        wrapper.setState({ ...newState });
        const updatedWrapper = wrapper.dive();

        it('matches snapshot', () => {
            expect(updatedWrapper).toMatchSnapshot();
        });

        it('renders <SignalsTable />', () => {
            const signalsTable = updatedWrapper.findWhere(
                node => node.props().dataTest === 'signals-table',
            );

            expect(signalsTable.exists()).toBeTruthy();
        });
    });

    describe('state changes based on lifecyle events with error', () => {
        const errorState = {
            tableResults: {},
            error: true,
            hasSearched: true,
        };

        wrapper.setState({ ...errorState });

        const wrapperWithError = wrapper.dive();

        it('matches snapshot', () => {
            expect(wrapperWithError).toMatchSnapshot();
        });

        it('renders <EmptySearch /> with "errorFetching" variant', () => {
            expect(wrapper.find(EmptySearch).exists()).toBeTruthy();
            expect(wrapper.find(EmptySearch).props().variant).toBe('errorFetching');
        });

        it('does not render <SignalsTable/>', () => {
            const signalsTable = wrapperWithError.findWhere(
                node => node.props().dataTest === 'signals-table',
            );

            expect(signalsTable.exists()).toBeFalsy();
        });
    });
});
