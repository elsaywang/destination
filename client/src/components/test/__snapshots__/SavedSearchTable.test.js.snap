// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavedSearchTable/> component rendering matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wait />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-valuemax": 100,
      "aria-valuemin": 0,
      "aria-valuenow": undefined,
      "children": Array [
        <div
          className="spectrum-CircleLoader-track"
/>,
        <div
          className="spectrum-CircleLoader-fills"
>
          <div
                    className="spectrum-CircleLoader-fillMask1"
          >
                    <div
                              className="spectrum-CircleLoader-fillSubMask1"
                              style={Object {}}
                    >
                              <div
                                        className="spectrum-CircleLoader-fill"
                              />
                    </div>
          </div>
          <div
                    className="spectrum-CircleLoader-fillMask2"
          >
                    <div
                              className="spectrum-CircleLoader-fillSubMask2"
                              style={Object {}}
                    >
                              <div
                                        className="spectrum-CircleLoader-fill"
                              />
                    </div>
          </div>
</div>,
      ],
      "className": "spectrum-CircleLoader spectrum-CircleLoader--indeterminate",
      "role": "progressbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "spectrum-CircleLoader-track",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="spectrum-CircleLoader-fillMask1"
>
              <div
                            className="spectrum-CircleLoader-fillSubMask1"
                            style={Object {}}
              >
                            <div
                                          className="spectrum-CircleLoader-fill"
                            />
              </div>
</div>,
            <div
              className="spectrum-CircleLoader-fillMask2"
>
              <div
                            className="spectrum-CircleLoader-fillSubMask2"
                            style={Object {}}
              >
                            <div
                                          className="spectrum-CircleLoader-fill"
                            />
              </div>
</div>,
          ],
          "className": "spectrum-CircleLoader-fills",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="spectrum-CircleLoader-fillSubMask1"
                style={Object {}}
>
                <div
                                className="spectrum-CircleLoader-fill"
                />
</div>,
              "className": "spectrum-CircleLoader-fillMask1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="spectrum-CircleLoader-fill"
/>,
                "className": "spectrum-CircleLoader-fillSubMask1",
                "style": Object {},
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "spectrum-CircleLoader-fill",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="spectrum-CircleLoader-fillSubMask2"
                style={Object {}}
>
                <div
                                className="spectrum-CircleLoader-fill"
                />
</div>,
              "className": "spectrum-CircleLoader-fillMask2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="spectrum-CircleLoader-fill"
/>,
                "className": "spectrum-CircleLoader-fillSubMask2",
                "style": Object {},
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "spectrum-CircleLoader-fill",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-valuemax": 100,
        "aria-valuemin": 0,
        "aria-valuenow": undefined,
        "children": Array [
          <div
            className="spectrum-CircleLoader-track"
/>,
          <div
            className="spectrum-CircleLoader-fills"
>
            <div
                        className="spectrum-CircleLoader-fillMask1"
            >
                        <div
                                    className="spectrum-CircleLoader-fillSubMask1"
                                    style={Object {}}
                        >
                                    <div
                                                className="spectrum-CircleLoader-fill"
                                    />
                        </div>
            </div>
            <div
                        className="spectrum-CircleLoader-fillMask2"
            >
                        <div
                                    className="spectrum-CircleLoader-fillSubMask2"
                                    style={Object {}}
                        >
                                    <div
                                                className="spectrum-CircleLoader-fill"
                                    />
                        </div>
            </div>
</div>,
        ],
        "className": "spectrum-CircleLoader spectrum-CircleLoader--indeterminate",
        "role": "progressbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "spectrum-CircleLoader-track",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="spectrum-CircleLoader-fillMask1"
>
                <div
                                className="spectrum-CircleLoader-fillSubMask1"
                                style={Object {}}
                >
                                <div
                                                className="spectrum-CircleLoader-fill"
                                />
                </div>
</div>,
              <div
                className="spectrum-CircleLoader-fillMask2"
>
                <div
                                className="spectrum-CircleLoader-fillSubMask2"
                                style={Object {}}
                >
                                <div
                                                className="spectrum-CircleLoader-fill"
                                />
                </div>
</div>,
            ],
            "className": "spectrum-CircleLoader-fills",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="spectrum-CircleLoader-fillSubMask1"
                  style={Object {}}
>
                  <div
                                    className="spectrum-CircleLoader-fill"
                  />
</div>,
                "className": "spectrum-CircleLoader-fillMask1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="spectrum-CircleLoader-fill"
/>,
                  "className": "spectrum-CircleLoader-fillSubMask1",
                  "style": Object {},
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "spectrum-CircleLoader-fill",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="spectrum-CircleLoader-fillSubMask2"
                  style={Object {}}
>
                  <div
                                    className="spectrum-CircleLoader-fill"
                  />
</div>,
                "className": "spectrum-CircleLoader-fillMask2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="spectrum-CircleLoader-fill"
/>,
                  "className": "spectrum-CircleLoader-fillSubMask2",
                  "style": Object {},
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "spectrum-CircleLoader-fill",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<SavedSearchTable/> component state changes based on lifecyle events with error matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptySearch
    className="noResults"
    variant="errorFetching"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <img
        alt="Error fetching data. Please try again later."
        className="noResultImage"
        data-test="error-fetching-data"
        src="localhost:3000/noResult.svg"
/>,
      "className": "noResults",
      "message": "Please try again later.",
      "messageStyle": "noResultMessage",
      "title": "Error fetching data.",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Error fetching data. Please try again later.",
        "className": "noResultImage",
        "data-test": "error-fetching-data",
        "src": "localhost:3000/noResult.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <img
          alt="Error fetching data. Please try again later."
          className="noResultImage"
          data-test="error-fetching-data"
          src="localhost:3000/noResult.svg"
/>,
        "className": "noResults",
        "message": "Please try again later.",
        "messageStyle": "noResultMessage",
        "title": "Error fetching data.",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Error fetching data. Please try again later.",
          "className": "noResultImage",
          "data-test": "error-fetching-data",
          "src": "localhost:3000/noResult.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<SavedSearchTable/> component state changes based on lifecyle events without error matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignalsTable
    allowsSelection={false}
    canCreateTraits={undefined}
    isAdvancedSearchEnabled={false}
    isLoaded={1}
    results={
        Array [
            Object {
              "categoryType": "REALTIME",
              "includedInTraits": null,
              "keyValuePairs": Array [
                Object {
                  "key": "evar18",
                  "value": "CMS18",
                },
              ],
              "percentageChange": 0.27783491545226535,
              "signalStatus": "UNUSED",
              "source": Object {
                "dataSourceIds": null,
                "reportSuiteIds": null,
                "sourceType": null,
              },
              "totalCount": 93859,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allowsSelection": false,
      "columns": Array [
        Object {
          "key": "keyValuePairs",
          "minWidth": 150,
          "sortable": true,
          "title": "Key-Value Pair",
        },
        Object {
          "key": "signalType",
          "title": "Signal Type",
          "width": 140,
        },
        Object {
          "key": "signalSource",
          "title": "Signal Source",
          "width": 180,
        },
        Object {
          "key": "totalCount",
          "sortable": true,
          "title": "Total Counts",
          "width": 96,
        },
        Object {
          "key": "percentageChange",
          "sortable": true,
          "title": "Percentage Change",
          "width": 165,
        },
        Object {
          "key": "includedInTraits",
          "title": "Included In Traits",
          "width": 155,
        },
      ],
      "dataTest": "signals-table",
      "isMaxSelectedRowsReached": undefined,
      "items": Array [
        Object {
          "categoryType": "REALTIME",
          "includedInTraits": Object {
            "categoryType": "REALTIME",
            "keyValuePairs": Array [
              Object {
                "key": "evar18",
                "value": "CMS18",
              },
            ],
            "sids": Array [],
            "source": Object {
              "dataSourceIds": null,
              "reportSuiteIds": null,
              "sourceType": null,
            },
          },
          "keyName": Array [
            "—",
          ],
          "keyValuePairs": Array [
            Object {
              "key": "evar18",
              "value": "CMS18",
            },
          ],
          "percentageChange": 0.27783491545226535,
          "signalSource": "—",
          "signalStatus": "UNUSED",
          "signalType": "—",
          "source": Object {
            "dataSourceIds": null,
            "reportSuiteIds": null,
            "sourceType": null,
          },
          "totalCount": 93859,
        },
      ],
      "onLoadMore": undefined,
      "onSelectionChange": [Function],
      "renderCell": [Function],
      "rowHeight": 48,
      "selectedRowIndexes": undefined,
      "sortSearch": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowsSelection": false,
        "columns": Array [
          Object {
            "key": "keyValuePairs",
            "minWidth": 150,
            "sortable": true,
            "title": "Key-Value Pair",
          },
          Object {
            "key": "signalType",
            "title": "Signal Type",
            "width": 140,
          },
          Object {
            "key": "signalSource",
            "title": "Signal Source",
            "width": 180,
          },
          Object {
            "key": "totalCount",
            "sortable": true,
            "title": "Total Counts",
            "width": 96,
          },
          Object {
            "key": "percentageChange",
            "sortable": true,
            "title": "Percentage Change",
            "width": 165,
          },
          Object {
            "key": "includedInTraits",
            "title": "Included In Traits",
            "width": 155,
          },
        ],
        "dataTest": "signals-table",
        "isMaxSelectedRowsReached": undefined,
        "items": Array [
          Object {
            "categoryType": "REALTIME",
            "includedInTraits": Object {
              "categoryType": "REALTIME",
              "keyValuePairs": Array [
                Object {
                  "key": "evar18",
                  "value": "CMS18",
                },
              ],
              "sids": Array [],
              "source": Object {
                "dataSourceIds": null,
                "reportSuiteIds": null,
                "sourceType": null,
              },
            },
            "keyName": Array [
              "—",
            ],
            "keyValuePairs": Array [
              Object {
                "key": "evar18",
                "value": "CMS18",
              },
            ],
            "percentageChange": 0.27783491545226535,
            "signalSource": "—",
            "signalStatus": "UNUSED",
            "signalType": "—",
            "source": Object {
              "dataSourceIds": null,
              "reportSuiteIds": null,
              "sourceType": null,
            },
            "totalCount": 93859,
          },
        ],
        "onLoadMore": undefined,
        "onSelectionChange": [Function],
        "renderCell": [Function],
        "rowHeight": 48,
        "selectedRowIndexes": undefined,
        "sortSearch": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
